File: src/main/java/base/Listeners/PlayerDeathListener.java
// src/main/java/me/stavgordeev/plugin/listeners/PlayerDeathListener.java
package base.Listeners;

import base.Minigames.DiscoMayhem.DiscoMayhem;
import base.Minigames.HoleInTheWall.HoleInTheWall;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

public class PlayerDeathListener implements Listener {
    private final DiscoMayhem discoMayhem;
    private final HoleInTheWall holeInTheWall;

    public PlayerDeathListener(DiscoMayhem discoMayhem, HoleInTheWall holeInTheWall) {
        this.discoMayhem = discoMayhem;
        this.holeInTheWall = holeInTheWall;
    }

    @EventHandler
    public void onPlayerDeath(PlayerDeathEvent event) {
        if (discoMayhem.isPlayerInGame(event.getEntity())) {
            discoMayhem.endGame();
        }
        if (holeInTheWall.isPlayerInGame(event.getEntity())) {
            holeInTheWall.endGame();
        }
    }
}

File: src/main/java/base/listeners/PlayerDeathListener.kt
// src/main/java/me/stavgordeev/plugin/listeners/PlayerDeathListener.java
package base.listeners

import base.minigames.disco_mayhem.DiscoMayhem
import base.minigames.hole_in_the_wall.HoleInTheWall
import base.minigames.maze_hunt.MazeHunt
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

class PlayerDeathListener(
    private val discoMayhem: DiscoMayhem,
    private val holeInTheWall: HoleInTheWall,
    private val mazeHunt: MazeHunt
) : Listener {
    @EventHandler
    fun onPlayerDeath(event: PlayerDeathEvent) {
        when {
            mazeHunt.isPlayerInGame(event.entity) -> mazeHunt.endGame()
            discoMayhem.isPlayerInGame(event.entity) -> discoMayhem.endGame()
            holeInTheWall.isPlayerInGame(event.entity) -> holeInTheWall.endGame()
        }
    }
}